sugar/helpers.py

```diff
@@ -65,7 +68,63 @@ def apply_slippage(amount: int, slippage: float) -> int:
     if slippage < 0 or slippage > 1: raise ValueError("Slippage must be between 0 and 1")
     return int(math.ceil(amount * (1 - slippage)))
 
+# %% ../src/helpers.ipynb 9
+def parse_ether(ether: str) -> int:
+    # Set precision high enough to handle 18 decimal places
+    getcontext().prec = 50
+    
+    try:
+        # Convert to Decimal for precise arithmetic
+        ether_decimal = Decimal(str(ether))
+        
+        # Convert to wei (multiply by 10^18)
+        wei_decimal = ether_decimal * Decimal('1000000000000000000')
+        
+        # Convert to integer
+        return int(wei_decimal)
+    
+    except (ValueError, TypeError, decimal.InvalidOperation) as e:
+        raise ValueError(f"Invalid ether value: {ether}") from e
+
+# %% ../src/helpers.ipynb 10
+def get_unique_str(length: int) -> str:
+    """
+    Generate a cryptographically secure random string of specified length.
+    
+    This is the Python equivalent of the TypeScript function using crypto.getRandomValues().
+    Uses secrets module for cryptographic security.
+    
+    Args:
+        length: Desired length of the random string
+        
+    Returns:
+        str: Random string of specified length containing digits
+        
+    Examples:
+        >>> len(get_unique_str(10))
+        10
+        >>> get_unique_str(5).isdigit()
+        True
+    """
+    # Generate random bytes and convert to string of digits
+    random_bytes = secrets.token_bytes(length)
+    return ''.join(str(byte % 10) for byte in random_bytes)[:length]
+
+def get_salt() -> str: return f"0x{get_unique_str(64)}" 
+
 # %% ../src/helpers.ipynb 12
+def to_bytes32(val: str) -> bytes: 
+    # Remove 0x prefix and pad to 64 hex characters (32 bytes)
+    hex_val = val.replace('0x', '').zfill(64)
+    return bytes.fromhex(hex_val)
+
+def to_bytes32_str(val: str) -> str: return f"0x{to_bytes32(val).hex()}"
+
+# %% ../src/helpers.ipynb 13
+# TODO: move this
+OPEN_USDT_TOKEN = normalize_address("0x1217bfe6c773eec6cc4a38b5dc45b92292b6e189")
```

Comment:
We should probably rename this to something more generic and make it chain specific.

Eg. the oUSDT will bridge into USDT on Celo.